modules/gapi/perf/common/gapi_video_perf_tests_inl.hpp

```diff
@@ -241,6 +241,151 @@ PERF_TEST_P_(BackgroundSubtractorPerfTest, TestPerformance)
+    ctrls.reserve(testNumMeasurements);
+    haveMeasurements.reserve(testNumMeasurements);
+    cv::RNG& rng = cv::theRNG();
+    {
+        cv::Mat measurement(mDim, 1, type);
+        cv::Mat ctrl(cDim, 1, type);
+        for (std::size_t i = 0; i < testNumMeasurements; i++)
+        {
+            haveMeasurements[i] = rng(2u) == 1; // returns 0 or 1 - whether we have measurement
+                                                // at this iteration or not
+            if (haveMeasurements[i])
+            {
+                cv::randu(measurement, cv::Scalar::all(-1), cv::Scalar::all(1));
+                measurements.push_back(measurement);
+            }
+            else
+            {
+                measurements.push_back(cv::Mat::zeros(mDim, 1, type));
+            }
+            cv::randu(ctrl, cv::Scalar::all(-1), cv::Scalar::all(1));
+            ctrls.push_back(ctrl);
+        }
+    }
+
+    // G-API graph declaration
+    cv::GMat m, ctrl;
+    cv::GOpaque<bool> have_m;
+    cv::GMat out = cv::gapi::KalmanFilter(m, have_m, ctrl, kp);
+    cv::GComputation c(cv::GIn(m, have_m, ctrl), cv::GOut(out));
+    auto cc = c.compile(
+        cv::descr_of(cv::gin(cv::Mat(mDim, 1, type), true, cv::Mat(cDim, 1, type))),
+        std::move(compileArgs));
+
+    cv::Mat gapiKState(dDim, 1, type);
+    TEST_CYCLE()
+    {
+        cc.prepareForNewStream();
+        for (std::size_t i = 0; i < testNumMeasurements; i++)
+        {
+            bool hvMeas = haveMeasurements[i];
+            cc(cv::gin(measurements[i], hvMeas, ctrls[i]), cv::gout(gapiKState));
+        }
+    }
+
+    // OpenCV reference KalmanFilter initialization
+    cv::KalmanFilter ocvKalman(dDim, mDim, cDim, type);
+    initKalmanFilter(kp, ocvKalman, true);
+
+    cv::Mat ocvKState(dDim, 1, type);
+    for (std::size_t i = 0; i < testNumMeasurements; i++)
+    {
+        ocvKState = ocvKalman.predict(ctrls[i]);
+        if (haveMeasurements[i])
+            ocvKState = ocvKalman.correct(measurements[i]);
+    }
+    // Validation
+    EXPECT_TRUE(AbsExact().to_compare_f()(gapiKState, ocvKState));
+    SANITY_CHECK_NOTHING();
+}
+
+PERF_TEST_P_(KalmanFilterNoControlPerfTest, TestPerformance)
+{
+    MatType2 type = -1;
+    int  dDim = -1, mDim = -1;
+    std::size_t testNumMeasurements = 0;
+    cv::GCompileArgs compileArgs;
+    std::tie(type, dDim, mDim, testNumMeasurements, compileArgs) = GetParam();
+
+    const int cDim = 0;
+    cv::gapi::KalmanParams kp;
+    initKalmanParams(kp, type, dDim, mDim, cDim);
+
+    // Generating input
+    std::vector<bool> haveMeasurements(testNumMeasurements);
+    std::vector<cv::Mat> measurements;
+    measurements.reserve(testNumMeasurements);
+    cv::RNG& rng = cv::theRNG();
+    {
+        cv::Mat measurement(mDim, 1, type);
+        for (std::size_t i = 0; i < testNumMeasurements; i++)
+        {
+            haveMeasurements[i] = rng(2u) == 1; // returns 0 or 1 - whether we have measurement
```

Comment:
Random value. Will tests have the same performance result in this case for next try?

Reply:
As it's `cv::`-random, I believe yes, they are exact. It's the same mechanism with a random `Mat` filling, and they work exact every try

Reply:
It is about possible case, when haveMeasurements[i] will equal:
flase, false, fasle, false...
and next launch:
true, true, false, true.
Will performance change for each launch? 
M b, it should be hardcoded or be a test parameter.

Reply:
@mpashchenkov your point is valid. Added a case when all the `haveMeasurements[i] == true` to test the slowest case scenario too.
Also I squashed all the reviewed commits into one. For your convenience, made a separate commit with the latest changes (fixing this issue)